lib_eio_linux/sched.ml

```diff
@@ -498,6 +498,26 @@ let config ?(queue_depth=64) ?n_blocks ?(block_size=4096) ?polling_timeout () =
 
 external eio_eventfd : int -> Unix.file_descr = "caml_eio_eventfd"
 
+let exit_uring uring =
+  (* Normally, all operations should have finished by the time we exit because
+     we don't exit until all fibers have finished, and a fiber can't finish while
+     an operation is in progress. However, this is not the case for cancellation
+     operations, which may still be active in rare cases. Drain them here. *)
+  let rec aux errors =
+    if Uring.active_ops uring = 0 then errors
+    else (
+      match Uring.wait ~timeout:1.0 uring with
```

Comment:
I'm not so sure about a 1s timeout here.  I could imagine an inflight operation taking longer than that (slow disk or net).  Is there any use in having a timeout here at all? It might be better just to leave the process hanging if there's something weird going on with Uring requests not completing.  It won't be deadlocked, since in theory the kernel can unwedge the process by pushing a completion event...

Reply:
I'm just wary of magic timeout numbers (the 1s).   Looping indefinitely with a log seems better. In a stress test, it's kind of hard to spot log entries with errors (although there would be a non-zero exit code here), but quite easy to spot hanging processes taking up all the room on the machine...